### PR TITLE
Event/hcai/leave event

### PR DESCRIPTION
**Pull request by TCbacon**
Leave event endpoint implementation

### CODE REVIEW

File: APILambda/events/leaveEvent.test.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  callCreateEvent,
+  callJoinEvent,
+  callLeaveEvent
+} from "../test/apiCallers/events.js"
+import { testEvent } from "../test/testEvents.js"
+import { createUserFlow } from "../test/userFlows/users.js"
+
+describe("Leave event tests", () => {
+  it("should return 204 if user leaves the event", async () => {
+    const { token: eventOwnerToken } = await createUserFlow()
```
Comment (seanim0920) [inline APILambda/events/leaveEvent.test.ts]: use createEventFlow()

File: APILambda/events/leaveEvent.test.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    const futureDate = new Date()
+    futureDate.setFullYear(futureDate.getFullYear() + 1)
+    const event = await callCreateEvent(eventOwnerToken, testEvent)
+    const resp = await callLeaveEvent(eventOwnerToken, event.body.id)
+
+    expect(resp).toMatchObject({
+      status: 400,
+      body: { error: "co-host-not-found" }
+    })
+  })
```
Comment (seanim0920) [inline APILambda/events/leaveEvent.test.ts:41]: TODO: Test eventAttendance endpoint to check that the user is no longer there

File: APILambda/events/leaveEvent.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  )
+
+// TODO: Handle adding co-host to event if main host decides to leave
+const leaveEvent = (conn: SQLExecutable, userId: string, eventId: number) => {
+  return conn.transaction((tx) =>
+    isHostUserFromOwnEvent(tx, userId, eventId)
+      .inverted()
+      .flatMapSuccess(() =>
+        removeUserFromAttendeeList(tx, userId, eventId).flatMapSuccess(
+          ({ rowsAffected }) => (rowsAffected > 0 ? success(204) : success(200))
```
Comment (seanim0920) [inline APILambda/events/leaveEvent.ts]: Add test for 200 case

File: APILambda/events/leaveEvent.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+) => {
+  /**
+   * Join an event
+   */
+  router.postWithValidation(
+    "/leave/:eventId",
+    { pathParamsSchema: joinEventSchema },
+    (req, res) =>
+      leaveEvent(conn, res.locals.selfId, Number(req.params.eventId))
+        .mapSuccess(() => res.status(204).json())
```
Comment (seanim0920) [inline APILambda/events/leaveEvent.ts]: should use parameterized status code

**Review by seanim0920 — DISMISSED** [review]

File: APILambda/events/leaveEvent.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  return conn.transaction((tx) =>
+    isHostUserFromOwnEvent(tx, userId, eventId)
+      .inverted()
+      .flatMapSuccess(() =>
+        removeUserFromAttendeeList(tx, userId, eventId).flatMapSuccess(
+          ({ rowsAffected }) => {
+            return rowsAffected > 0 ? success(204) : success(200)
+          }
+        )
+      )
```
Comment (TCbacon) [inline APILambda/events/leaveEvent.ts]: move inner flatmapsuccess outside and use mapsuccess instead

**Review by seanim0920 — APPROVED** [review]

**Pull request merged**